src/main/java/org/tom_v_squad/soiwenttoaconcert/controllers/EventController.java

```diff
@@ -47,5 +45,81 @@ public String displayEventList(Model model) {
         return "events/index";
     }
 
+    @GetMapping("edit/{eventId}")
+    public String editEventForm(@PathVariable Integer eventId, Model model) {
+        Optional<Event> result = eventRepository.findById(eventId);
 
+        if (result.isEmpty()) {
+            model.addAttribute("title", "Invalid Event ID: " + eventId);
+        }else{
+            Event event = result.get();
+            //System.out.println(event.getEventId());
+            model.addAttribute("event", event);
+        }
+
+        return "events/edit";
+
+    }
+
+    @PostMapping("edit/{eventId}")
+    public String editEventForm(@PathVariable Integer eventId, @ModelAttribute Event newEvent, Errors errors, Model model) {
+        if(errors.hasErrors()) {
+            model.addAttribute("title", "Create Event");
+            return "events/create";
+        }
+       // model.addAttribute("location", location);
+        eventRepository.save(newEvent);
+        return "events/index";
+    }
+
+    @GetMapping("delete/{eventId}")
+//    @GetMapping("delete")
+    public String displayDeleteEventForm(@RequestParam Integer eventId, Model model) {
+//        Optional<Event> result = eventRepository.findById(eventId);
+////attach a collection of events to loop thru + display
+//        if (!result.isEmpty()) {
+//            Event event = result.get();
+//            model.addAttribute("title","Delete Event");
+//            model.addAttribute("event", event);
+//            return "events/delete";
+//
+//        }else{
+//            model.addAttribute("title", "Invalid Event ID: " + eventId);
+//        }
+
+        return "events/delete";
+
+    }
+
+//    @GetMapping("delete")
+//    public String renderDeleteEventForm(Model model) {
+//        model.addAttribute("title", "Delete Event");
+//        model.addAttribute("events", eventRepository.findAll());
+//        return "events/delete";
+//    }
+
+    @PostMapping("delete")
+    public String processDeleteEventsForm(@RequestParam(required = false) Integer eventId) {
+
+        if (eventId != null) {
+                Optional<Event> result = eventRepository.findById(eventId);
+                eventRepository.delete(result.get());
```

Comment:
Prior to this you'll want to make sure that `result` contains a value - you can look at the `displayDeleteEventForm` method for an example of this.